### PR TITLE
Update federation docs to recommend supergraphSdl over serviceList

### PR DESCRIPTION
**Pull request by StephenBarlow**

### CODE REVIEW

**Comment by StephenBarlow** [comment]
@lennyburdette I recall your saying you wish `serviceList` would go away in the federation docs 😄

**Comment by prasek** [comment]
@StephenBarlow `supergraphSdl` should definitely be the recommended way in the docs, but we'll want to keep `serviceList` around for a while to ensure a graceful transition for everyone.

**Review by abernix — APPROVED** [review]

File: docs/source/migrating-from-stitching.md [hunk 1]
```diff
@@ -79,7 +79,7 @@ After you've registered your schemas, you can start exposing your subgraphs from
 
 We recommend setting up the Apollo Server gateway _alongside_ your existing schema-stitching gateway. Depending on your infrastructure, you might even want to run both in the same _process_ to support dynamically routing traffic through one gateway or the other.
 
-To enable managed configuration with Apollo Studio, set the `APOLLO_KEY` and `APOLLO_GRAPH_VARIANT` environment variables when you start up your Apollo Server gateway, and **do not provide the `serviceList` constructor option to `ApolloGateway`**. For details, see the [Apollo Studio documentation](https://www.apollographql.com/docs/studio/managed-federation/setup/).
+To enable managed configuration with Apollo Studio, set the `APOLLO_KEY` and `APOLLO_GRAPH_VARIANT` environment variables when you start up your Apollo Server gateway, and **do not provide the `supergraphSDL` or `serviceList` constructor option to `ApolloGateway`**. For details, see the [Apollo Studio documentation](https://www.apollographql.com/docs/studio/managed-federation/setup/).
```
Comment (trevor-scheer) [inline docs/source/migrating-from-stitching.md:82]:
```suggestion
To enable managed configuration with Apollo Studio, set the `APOLLO_KEY` and `APOLLO_GRAPH_REF` environment variables when you start up your Apollo Server gateway, and **do not provide the `supergraphSDL` or `serviceList` constructor option to `ApolloGateway`**. For details, see the [Apollo Studio documentation](https://www.apollographql.com/docs/studio/managed-federation/setup/).
```
The existence of this makes me wonder if the federation docs didn't get the update when we made this change for AS3.

(I looked, there's [one other](https://github.com/apollographql/federation/blob/main/docs/source/metrics.md) that needs an update, but this is unrelated to this PR)

**Review by trevor-scheer — APPROVED** [review]
